apps/myaccount/src/components/account-recovery/options/sms-recovery/sms-recovery.tsx

```diff
@@ -0,0 +1,429 @@
+import { useDispatch, useSelector } from "react-redux";
+import { AnyAction } from "redux";
+import { ThunkDispatch } from "redux-thunk";
+import { Form, Grid, Icon, List } from "semantic-ui-react";
+import { updateProfileInfo } from "../../../../api";
+import { getAccountRecoveryIcons } from "../../../../configs";
+import { CommonConstants } from "../../../../constants";
+import { commonConfig } from "../../../../extensions";
+import { AlertInterface, AlertLevels, BasicProfileInterface, MultiValue, ProfileSchema } from "../../../../models";
+import { AppState } from "../../../../store";
+import { getProfileInformation, setActiveForm } from "../../../../store/actions";
+import { EditSection } from "../../../shared";
+import "./sms-recovery.scss";
+
+/**
+ * SMS key.
+ */
+const SMS: string = "sms";
+
+/**
+ * Prop types for the SMSRecoveryComponent component.
+ * Also see {@link SMSRecovery.defaultProps}
+ */
+interface SMSRecoveryProps extends IdentifiableComponentInterface {
+    onAlertFired: (alert: AlertInterface) => void;
+    enableEditMobile?: boolean;
+}
+
+/**
+ * SMS recovery section.
+ *
+ * @param props - Props injected to the component.
+ * @returns React element.
+ */
+export const SMSRecovery: React.FunctionComponent<SMSRecoveryProps> = (
+    props: SMSRecoveryProps
+): ReactElement => {
+
+    const {
+        onAlertFired,
+        enableEditMobile,
+        ["data-componentid"]: componentId
+    } = props;
+
+    const { t } = useTranslation();
+    const dispatch: ThunkDispatch<AppState, any, AnyAction> = useDispatch();
+
+    const profileInfo: BasicProfileInterface = useSelector(
+        (state: AppState) => state.authenticationInformation.profileInfo
+    );
+    const phoneNumberSchema: ProfileSchema = useSelector((state: AppState) => {
+        const phoneNumberSchemas: ProfileSchema = state.authenticationInformation.profileSchemas.find(
+            (profileSchema: ProfileSchema) => profileSchema.name === "phoneNumbers");
+
+        if (phoneNumberSchemas && phoneNumberSchemas.subAttributes) {
+            const mobileSchema: ProfileSchema = phoneNumberSchemas.subAttributes?.find(
+                (subAttribute: ProfileSchema) => subAttribute.name === "mobile");
+
+            if (mobileSchema) return mobileSchema;
+        }
+
+        return phoneNumberSchemas;
+    });
+    const activeForm: string = useSelector((state: AppState) => state.global.activeForm);
+
+    const [ mobile, setMobile ] = useState<string>(undefined);
+    const [ editedMobile, setEditedMobile ] = useState<string>(undefined);
+
+    let mobileType: string;
+
+    useEffect(() => {
+        if (isEmpty(profileInfo)) {
+            dispatch(getProfileInformation());
+        }
+    }, []);
+
+    const handleUpdate = (mobile: string) => {
+        const data: {
+            Operations: {
+                op: string;
+                value: Record<string, unknown>;
+            }[];
+            schemas: string[];
+        } = {
```

Comment:
AFAIR, there is already an interface defined for this. (something like PatchInterface). Could u please check and update?

Reply:
Currently there is no interface for this. All places that use this interface use an in place defined interface.